ocaml/gencert/selfcert.ml

```diff
@@ -120,16 +120,17 @@ let generate_pub_priv_key length =
     |> X509.Private_key.decode_pem
     |> R.reword_error (fun _ -> R.msg "decoding private key failed")
   in
+  let err_not_rsa = R.error_msg "generated private key does not use RSA" in
   let* rsa =
-    try match privkey with `RSA x -> Ok x
-    with _ -> R.error_msg "generated private key does not use RSA"
+    try match privkey with `RSA x -> Ok x | _ -> err_not_rsa
+    with _ -> err_not_rsa
```

Comment:
Is the `try` really needed here? If you handle the wildcard in the inner match then the match itself cannot raise exceptions, and there is nothing else there to raise exceptions.